source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -541,11 +541,32 @@ namespace AdaptiveCards::Rendering::Uwp::ActionHelpers
                     tooltip = title;
                 }
             }
-            else
+            else if (!tooltip.empty())
             {
                 // If we don't have a title, use the tooltip as the name
                 name = tooltip;
             }
+            else
+            {
+                // There is an edge case here where the column is being used as a toggle button,
+                // i.e. it contains 2 items, one visible and one invisibile, and they swap visibilities
+                // upon the column being clicked. In this case, we need the AltText of the visible item
+                // since the column itself does not have one
+                if (const auto col = adaptiveCardElement.try_as<winrt::AdaptiveColumn>())
+                {
+                    for (auto item : col.Items())
+                    {
+                        if (item.IsVisible())
+                        {
+                            if (const auto img = item.try_as<winrt::AdaptiveImage>())
+                            {
+                                name = img.AltText();
+                                tooltip = img.AltText();
+                            }
+                        }
+                    }
+                }
```

Comment:
I am unsure about this - can this be generalized? This code assumes the specific case where the `adaptiveCardElement` is an `AdaptiveColumn` and that its items are of the type `AdaptiveImage`. However I could imagine a scenario where we have the same problem but with a different type of Container and different type of Item.

Reply:
[ToggleVisibilty](https://adaptivecards.io/explorer/Action.ToggleVisibility.html) has property `targetElements` array.

It lists the elements whose visibility should be toggled when triggered. You could use the list and element's id in the list to figure out which will be visible and use it to grab the tool tip.